neo/io/nwbio.py

```diff
@@ -0,0 +1,810 @@
+        electrodes = self._write_electrodes(nwbfile, block)
+        if not block.name:
+            block.name = "block%d" % self.blocks_written
+        for i, segment in enumerate(block.segments):
+            assert segment.block is block
+            if not segment.name:
+                segment.name = "%s : segment%d" % (block.name, i)
+            self._write_segment(nwbfile, segment, electrodes)
+        self.blocks_written += 1
+
+    def _write_electrodes(self, nwbfile, block):
+        # this handles only icephys_electrode for now
+        electrodes = {}
+        devices = {}
+        for segment in block.segments:
+            for signal in chain(segment.analogsignals, segment.irregularlysampledsignals):
+                if "nwb_electrode" in signal.annotations:
+                    elec_meta = signal.annotations["nwb_electrode"].copy()
+                    if elec_meta["name"] not in electrodes:
+                        # todo: check for consistency if the name is already there
+                        if elec_meta["device"]["name"] in devices:
+                            device = devices[elec_meta["device"]["name"]]
+                        else:
+                            device = nwbfile.create_device(**elec_meta["device"])
+                            devices[elec_meta["device"]["name"]] = device
+                        elec_meta.pop("device")
+                        electrodes[elec_meta["name"]] = nwbfile.create_icephys_electrode(
+                            device=device, **elec_meta
+                        )
+        return electrodes
+
+    def _write_segment(self, nwbfile, segment, electrodes):
+        # maybe use NWB trials to store Segment metadata?
+        for i, signal in enumerate(chain(segment.analogsignals, segment.irregularlysampledsignals)):
+            assert signal.segment is segment
+            if not signal.name:
+                signal.name = "%s : analogsignal%d" % (segment.name, i)
+            self._write_signal(nwbfile, signal, electrodes)
+
+        for i, train in enumerate(segment.spiketrains):
+            assert train.segment is segment
+            if not train.name:
+                train.name = "%s : spiketrain%d" % (segment.name, i)
+            self._write_spiketrain(nwbfile, train)
+
+        for i, event in enumerate(segment.events):
+            assert event.segment is segment
+            if not event.name:
+                event.name = "%s : event%d" % (segment.name, i)
+            self._write_event(nwbfile, event)
+
+        for i, epoch in enumerate(segment.epochs):
+            if not epoch.name:
+                epoch.name = "%s : epoch%d" % (segment.name, i)
+            self._write_epoch(nwbfile, epoch)
+
+    def _write_signal(self, nwbfile, signal, electrodes):
+        hierarchy = {'block': signal.segment.block.name, 'segment': signal.segment.name}
+        if "nwb_type" in signal.annotations:
+            timeseries_class = get_class(*signal.annotations["nwb_type"])
```

Comment:
We call this "neurodata_type". I understand why you might want to call it "nwb_type" here, but just letting you know in case you wanted to be consistent with our naming convention

Reply:
Perhaps "nwb_neurodata_type"? That preserves the clarity of the "nwb_" prefix in a Neo context, but makes the NWB equivalency more evident.

Reply:
yeah I think that would work

Reply:
You are right. "nwb neurodata_type" seems to be clear to both NWB and Neo.

Reply:
seems to be implemented.